### DISCUSSION TITLE
Building a machine learning app with Blitz

### DISCUSSION THREAD

**Discussion by spidertwin2** [blitz-js/blitz#2136] [Q&A] [answered]
Hello,

I wanted advice on using Blitz to build an ML app that uses natural language processing.

- How should I set up Blitz to work with Python processes in the backend?
- Would Fauna be a good choice of database?
- What would be the recommended hosting platform for my use case?

Thank you :pray:

**Comment by Skn0tt** [comment, marked as answer]
*How should Blitz communicate with Python?* A lot of this depends on what part of the work should be performed in your python process. Do you want it to be responsible for _just_ the machine learning part? Or should it also deal with stuff like user profiles, sending emails, billing etc?
If it should do _just_ the machine learning part, then you could deploy it separately (I think the public clouds have some offerings specifically catered to deploying ML, right?) and make it available to your Blitz app via HTTP.
If it should also do user profiles etc., you should think about deploying them together on the same host and maybe use something like a job queue (is there a celery client for JS?) or plain IPC to communicate.

For *databases*, it really depends on your specific data requirements. "an ML app" is pretty vague - what kind of data do you process? Is it structured / unstructured, is it relational? I've heard not-so-great stuff about Fauna's pricing, but your mileage may vary.

Re *hosting providers*: It depends on the scale you're going for and wether you're comfortable doing some admin work yourself. In the beginning, renting a cheap VPS and running your DB, Blitz and Python processes on the same machine should be fine. Once there's some traffic, you can re-evaluate what hosting you want to go for.
I'm sure that other people in the community can give a more nuanced view on this :)

**Reply by spidertwin2** [reply]
This is really helpful! :evergreen_tree: 

I am going for - 

PostgreSQL - database
Prisma 2 - ORM
Vercel - host
